component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java

```diff
@@ -100,12 +101,36 @@ private void addComponentParameters() {
             SettingsCreator advancedCreator = new SettingsCreator(node, ADVANCED, updateComponentsParameter);
             root.accept(advancedCreator, Metadatas.ADVANCED_FORM);
             parameters.addAll(advancedCreator.getSettings());
-            // If No schema add one
-            if (parameters.stream().noneMatch(p -> EParameterFieldType.SCHEMA_TYPE.equals(p.getFieldType()))) {
-                parameters
-                        .add(mainSettingsCreator.createSchemaParameter(EConnectionType.FLOW_MAIN.getName(), "SCHEMA"));
-            }
+            checkSchemaProperties(node, mainSettingsCreator);
+        }
+    }
 
+    /**
+     * Check whether all required schema settings are created and create ones we still need depending on connectors we
+     * have.
+     * 
+     * @param node
+     * @param mainSettingsCreator
+     */
+    protected void checkSchemaProperties(final INode node, final SettingsCreator mainSettingsCreator) {
```

Comment:
seems, `node` parameter is not required here. It is already available as a class field

Reply:
fixed